src/syscheckd/db/include/db.hpp

```diff
@@ -55,6 +55,23 @@ void fim_db_init(int storage,
                  logging_callback_t log_callback);
 #endif
 
+/**
+ * @brief Push a message to the syscheck queue
+ *
+ * @param msg The specific message to be pushed
+ */
+void fim_sync_push_msg(const char *msg);
+
+/**
+ * @brief Thread that performs the syscheck data synchronization
+ *
+ */
+#ifdef WIN32
+DWORD WINAPI fim_run_integrity(void __attribute__((unused)) * args);
```

Comment:
We can create the thread from c++ (we also remove several ifdefs) using std::thread which is cross-platform. 
This way we don't have two interfaces and we abstract the operating system for synchronization.

Reply:
Done